api/src/lib.rs

```diff
@@ -642,6 +689,16 @@ impl Client {
             .chain_err(|| ErrorKind::BadJsonResponse)?
             .into_result(status)
     }
+
+    fn with_retries<F>(&self, f: F) -> ReqwestResult<HttpResponse>
```

Comment:
same nit on `f`, maybe `send_request` instead? I'm slightly less fussed about `F`, but even there we tend to do `SendRequestT`